src/components/FormStep.js

```diff
@@ -504,6 +670,19 @@ const FormStep = ({form, submission, onLogicChecked, onStepSubmitted, onLogout})
     history.push(navigateTo);
   };
 
+  /**
+   * Called when the form has completed the render, attach, and one initialization change event
+   * loop.
+   * @see {@link Form.io documentation} https://help.form.io/developers/form-renderer#form-events
+   *
+   *  - We cannot filter 'blank' values to prevent Formio validation from running, as Formio will
+   *    use the default values in that case which have been explicitly unset.
+   *  - In the situation that we have invalid backend data (loading a submission with a required
+   *    field with default value that was cleared, for example), we _need_ to see the validation
+   *    errors since the data is not valid.
+   *  - For the initial, empty form load, no validation errors are displayed as there is no
+   *    respective backend data.
```

Comment:
this really belongs in the context _close_ to where it's relevant, so please put it back as inline comment.